### ISSUE TITLE
cc.pix_fmt (pixel format) error

### ISSUE THREAD

**Issue by pure-maple** [WyattBlue/auto-editor#489]
### Bug description

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Maple\AppData\Roaming\Python\Python312\Scripts\auto-editor.exe\__main__.py", line 7, in <module>
  File "C:\Users\Maple\AppData\Roaming\Python\Python312\site-packages\auto_editor\__main__.py", line 336, in main
    edit_media(paths, ffmpeg, args, temp, log)
  File "C:\Users\Maple\AppData\Roaming\Python\Python312\site-packages\auto_editor\edit.py", line 176, in edit_media
    sources = [initFileInfo(path, log) for path in paths]
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Maple\AppData\Roaming\Python\Python312\site-packages\auto_editor\ffwrapper.py", line 235, in initFileInfo
    cc.pix_fmt,
    ^^^^^^^^^^
  File "av\\video\\codeccontext.pyx", line 126, in av.video.codeccontext.VideoCodecContext.pix_fmt.__get__
AttributeError: 'NoneType' object has no attribute 'name'
```

### What's your platform?

```shell
OS: Windows 11 amd64
Python: 3.12.2
PyAV: 12.3.0
Auto-Editor: 24w31a
```

### What command did you use

```shell
`auto-editor 1.mp4 --my-ffmpeg -c:v copy --audio-codec copy --margin 0.1sec --show-ffmpeg-commands --show-ffmpeg-output --no-open`;
`auto-editor 1.mp4`;
```

### What properties does your input video have?

```
ffmpeg version 2024-03-11-git-3d1860ec8d-full_build-www.gyan.dev Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 13.2.0 (Rev5, Built by MSYS2 project)
  configuration: --enable-gpl --enable-version3 --enable-static --pkg-config=pkgconf --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-bzlib --enable-lzma --enable-libsnappy --enable-zlib --enable-librist --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-libbluray --enable-libcaca --enable-sdl2 --enable-libaribb24 --enable-libaribcaption --enable-libdav1d --enable-libdavs2 --enable-libuavs3d --enable-libzvbi --enable-librav1e --enable-libsvtav1 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libaom --enable-libjxl --enable-libopenjpeg --enable-libvpx --enable-mediafoundation --enable-libass --enable-frei0r --enable-libfreetype --enable-libfribidi --enable-libharfbuzz --enable-liblensfun --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-dxva2 --enable-d3d11va --enable-libvpl --enable-libshaderc --enable-vulkan --enable-libplacebo --enable-opencl --enable-libcdio --enable-libgme --enable-libmodplug --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libshine --enable-libtheora --enable-libtwolame --enable-libvo-amrwbenc --enable-libcodec2 --enable-libilbc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-ladspa --enable-libbs2b --enable-libflite --enable-libmysofa --enable-librubberband --enable-libsoxr --enable-chromaprint
  libavutil      59.  1.100 / 59.  1.100
  libavcodec     61.  1.100 / 61.  1.100
  libavformat    61.  0.100 / 61.  0.100
  libavdevice    61.  0.100 / 61.  0.100
  libavfilter    10.  0.100 / 10.  0.100
  libswscale      8.  0.100 /  8.  0.100
  libswresample   5.  0.100 /  5.  0.100
  libpostproc    58.  0.100 / 58.  0.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 000002661aeda500] Could not find codec parameters for stream 0 (Video: h264 (avc1 / 0x31637661), none, 1920x1080, 395 kb/s): unspecified pixel format
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.76.100
  Duration: 01:58:58.11, start: 0.000000, bitrate: 432 kb/s
  Stream #0:0[0x1](und): Video: h264 (avc1 / 0x31637661), none, 1920x1080, 395 kb/s, 12.47 fps, 90k tbr, 90k tbn (default)
      Metadata:
        handler_name    : VideoHandler
        vendor_id       : [0][0][0][0]
  Stream #0:1[0x2](und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, mono, fltp, 61 kb/s (default)
      Metadata:
        handler_name    : SoundHandler
        vendor_id       : [0][0][0][0]
At least one output file must be specified
```

**Comment by WyattBlue** [comment]
This is really a PyAV issue. I've fixed this here https://github.com/PyAV-Org/PyAV/pull/1488. You can compile PyAV yourself or  wait for PyAV 13 in September

**Comment by pure-maple** [comment]
> This is really a PyAV issue. I've fixed this here [PyAV-Org/PyAV#1488](https://github.com/PyAV-Org/PyAV/pull/1488). You can compile PyAV yourself or wait for PyAV 13 in September

How should I compile it, and what command should I use to build after cloning the repository?

**Comment by pure-maple** [comment]
> This is really a PyAV issue. I've fixed this here [PyAV-Org/PyAV#1488](https://github.com/PyAV-Org/PyAV/pull/1488). You can compile PyAV yourself or wait for PyAV 13 in September

I have cloned the auto-editor and PyAV repositories. I used `pip install -e .` to install auto-editor and `python setup.py install` to install PyAV. However, when I run the auto-editor command again, it still displays information like this:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Maple\PycharmProjects\AudioVideo\.venv\Scripts\auto-editor.exe\__main__.py", line 7, in <module>
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\__main__.py", line 329, in main
    edit_media(paths, ffmpeg, args, log)
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\edit.py", line 174, in edit_media
    sources = [initFileInfo(path, log) for path in paths]
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\ffwrapper.py", line 235, in initFileInfo
    cc.pix_fmt,
    ^^^^^^^^^^
  File "av\\video\\codeccontext.pyx", line 126, in av.video.codeccontext.VideoCodecContext.pix_fmt.__get__
AttributeError: 'NoneType' object has no attribute 'name'
```
Are my steps correct? Thank you for your help.

**Comment by pure-maple** [comment]
> This is really a PyAV issue. I've fixed this here [PyAV-Org/PyAV#1488](https://github.com/PyAV-Org/PyAV/pull/1488). You can compile PyAV yourself or wait for PyAV 13 in September

**Feedback on cc.pix_fmt Error and Solutions**

Hi,

I’d like to share some steps I took to address the `cc.pix_fmt` issue:

1. Cloned the PyAV repository, built it using `python setup.py build --ffmpeg-dir=D:\DevTools\ffmpeg`, and installed it with `python setup.py install`.
2. Cloned the auto-editor repository, then modified `auto-editor/auto_editor/ffwrapper.py` as follows:

```python
pix_fmt = None
if cc is not None:
    try:
        pix_fmt = cc.pix_fmt
    except AttributeError:
        pix_fmt = None

videos += (
    VideoStream(
        v.width,
        v.height,
        v.name,
        fps,
        vdur,
        sar,
        v.time_base,
        pix_fmt,
        cc.color_range,
        cc.colorspace,
        cc.color_primaries,
        cc.color_trc,
        0 if v.bit_rate is None else v.bit_rate,
        v.language,
    ),
)
```

3. Built auto-editor with `pip install -e .`.
4. The program ran successfully after these changes.

I also tried modifying `PyAV/av/video/codeccontext.pyx` as follows:

```python
@property
def pix_fmt(self):
    """
    The pixel format's name.

    :type: str | None
    """
    if self._format is None:
        return None
    return getattr(self._format, "name", None)
```

However, this change didn’t resolve the issue. Is the `try-except` method in auto-editor the correct approach, or is there a better way to handle this in PyAV? Additionally, is it possible to install PyAV using `pip install -e .`?

**Comment by pure-maple** [comment]
After building PyAV and using auto-editor, I encountered the following error:

```
  File "C:\Users\Maple\PycharmProjects\AudioVideo\.venv\Lib\site-packages\av-13.0.0rc1-py3.12-win-amd64.egg\av\__init__.py", line 20, in <module>
    from av._core import time_base, library_versions
ImportError: DLL load failed while importing _core: The specified module could not be found.
```

To resolve this, I manually added `os.add_dll_directory(r"D:\DevTools\ffmpeg\bin")` in `__init__.py`, which allowed the program to run successfully and resolved the `'hevc_nvenc' codec is not supported in 'mp4' container` issue #490 .

However, I also faced an `AttributeError`:

```
Traceback (most recent call last):                                                                                                                                                                                                                  
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\Maple\PycharmProjects\AudioVideo\.venv\Scripts\auto-editor.exe\__main__.py", line 7, in <module>
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\__main__.py", line 329, in main
    edit_media(paths, ffmpeg, args, log)
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\edit.py", line 338, in edit_media
    make_media(tl, output)
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\edit.py", line 274, in make_media
    out_path, apply_later = render_av(ffmpeg, tl, args, bar, ctr, log)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Maple\PycharmProjects\AudioVideo\auto-editor\auto_editor\render\video.py", line 145, in render_av
    if src == first_src and stream.pix_fmt is not None:
                            ^^^^^^^^^^^^^^
  File "av\\video\\stream.pyx", line 21, in av.video.stream.VideoStream.__getattr__
  File "av\\video\\codeccontext.pyx", line 123, in av.video.codeccontext.VideoCodecContext.pix_fmt.__get__
AttributeError: 'NoneType' object has no attribute 'name'
```

I resolved this issue by modifying `PyAV/av/video/codeccontext.pyx` as follows:

```python
@property
def pix_fmt(self):
    """
    The pixel format's name.

    :type: str | None
    """
    if self._format is None:
        return None
    return getattr(self._format, "name", None)
```

This resolved the `AttributeError`. However, I'm looking for guidance on how to properly handle the DLL import issue without manually modifying the `__init__.py` file.
